### ISSUE TITLE
how to create Allowances  And Charges?

### ISSUE THREAD

**Issue by homebeaver** [klst-de/e-invoice#25]
need an example.

**Comment by homebeaver** [comment]
here some examples:
- simple LINE ALLOWANCES, any line can add an allowence or charge
- LINE CHARGES with (optional) reason and reason text: any line has a factory `create `method 

```java
		// BG-27 0..n LINE ALLOWANCES:
BigDecimal tenPerCent = new BigDecimal(10);
line.addAllowance(new Amount(new BigDecimal(6.00)), new Amount(new BigDecimal(60.00)), tenPerCent);
		// BG-28 0..n LINE CHARGES:
AllowancesAndCharges charge = line.createCharge(new Amount(new BigDecimal(6.00)), new Amount(new BigDecimal(60.00)), tenPerCent);
charge.setReasoncode("64");                // UNTDID 5189 code
charge.setReasonText("Special agreement");
line.addAllowanceCharge(charge);

```

**Comment by homebeaver** [comment]
solved in 3.1.0
